

:date: 2013-10-10 23:27:09
:tags: проблема, Django, транзакция, 1.4, MyISAM, python, MySQL, InnoDB
:category: text
:author: IMDagger
:yaru-link: http://imdagger.ya.ru/replies.xml?item_no=1874
:slug: urn:ya.ru:post/22199227/1874

Если есть по наследству Django 1.4 (как у меня), то лучше не забыть
и перепроверить, а указан ли параметр STORAGE\_ENGINE для MySQL. Иначе
вдруг окажется, что работаем с MySQL версии 5.1 под CentOS (как в моём
случае) и там по умолчанию движок MyISAM, который не поддерживает
транзакции. Что может оказаться сюрпризом, когда :code:`with
transaction.commit\_on\_success()` не срабатывает ожидаемым образом.
Поэтому пришлось прописать :code:`'STORAGE\_ENGINE': 'INNODB'` и сконвертировать
старые таблицы используя :code:`ALTER TABLE \`таблица\` ENGINE = InnoDB;` Можно
было и настройки MySQL поменять, но лучше конечно же перестраховаться, и
в коде тоже выставить требование по движку базы.
